htdocs/product/class/product.class.php

```diff
@@ -2643,7 +2652,10 @@ public function load_stats_commande($socid = 0, $filtrestatut = '', $forVirtualS
                     $this->stats_commande['qty'] -= $adeduire;
                 }
             }
-
+            $hookmanager->initHooks(array('productdao'));
```

Comment:
productdao does not seems to be a page call context. Because you get hookmanager from an already loaded variable (loaded by calling main page), there is no reason to init here another hook context. 
And if you need your hook to be called each time the load_stats_xxx is called, you arleady have the context "main" that is defined.